### PR TITLE
Fix bug with repeated list event.

### PR DESCRIPTION
**Pull request by prabhuramachandran**
Fixing an issue with threads sending events.  Earlier, an RLock was used but this is not enough.  tornado does not support multiple threads calling send_message etc.  Instead one should add a callback on the IOLoop instance.  See: 

http://www.tornadoweb.org/en/stable/web.html?highlight=thread#thread-safety-notes

Also switch to using the Chrome driver on Darwin.  Newer versions of
Firefox (>45) do not work with selenium.

### CODE REVIEW

**Comment by prabhuramachandran** [comment]
@agrawalprash -- don't merge this yet, I think I found a proper fix.

**Comment by prabhuramachandran** [comment]
@agrawalprash -- this should fix the problem we were seeing but the current test error on travis is not reproducible on my mac and unrelated to this fix.

**Comment by prabhuramachandran** [comment]
@agrawalprash -- RTM.  The tests should be better behaved now.

File: jigna/tests/test_jigna_vue_web.py [hunk 1]
```diff
@@ -42,11 +43,17 @@ def setUpClass(cls, async=False):
         t.setDaemon(True)
         t.start()
 
-        browser = webdriver.Firefox()
+        if sys.platform.startswith('darwin'):
```
Comment (agrawalprash) [inline jigna/tests/test_jigna_vue_web.py]: Can you please add a comment describing the reason for switching to Chrome for OSX and maybe a link to the issue which says FF>45 doesn't work with selenium? Thanks.

File: jigna/web_server.py [hunk 2]
```diff
@@ -50,7 +54,10 @@ def send_event(self, event):
         message_id = -1
         data = json.dumps([message_id, jsonized_event])
         for socket in self._active_sockets:
-            socket.write_message(data)
+            if main_thread:
+                socket.write_message(data)
+            else:
```
Comment (agrawalprash) [inline jigna/web_server.py]: Maybe add a comment here that tornado does not support calling `socket.write_message` from a thread and a link to the article you mentioned on the PR comment?

**Pull request merged**